inc/Engine/Optimization/RUCSS/AbstractAPIClient.php

```diff
@@ -0,0 +1,48 @@
+<?php
+declare(strict_types=1);
+
+namespace WP_Rocket\Engine\Optimization\RUCSS;
+
+abstract class AbstractAPIClient {
+	/**
+	 * API URL.
+	 */
+	const API_URL = 'https://central-saas.wp-rocket.me/';
+
+	/**
+	 * SAAS main API path.
+	 */
+	const TREESHAKE_PATH = 'api';
+
+	/**
+	 * API Warmup path.
+	 */
+	const WARMUP_PATH = 'warmup';
+
+	/**
+	 * Handle Saas request error.
+	 *
+	 * @param  array|WP_Error $request WP Remote request.
+	 *
+	 * @return array
+	 */
+	protected function handle_request_error( $request ) {
```

Comment:
so this method will be called only on treeshaker, right?

I think we need to unify both APIClients to be able to use this method generically, what do u think?

Reply:
Yeah. In TreeShaker I return both the code and the error message why it could not tree shake the CSS.
Not sure what Warmup needs, but a message should be good for logging and debuging.